spec/std/string_spec.cr

```diff
@@ -1752,8 +1752,7 @@ describe "String" do
     end
 
     it "works with strings with block" do
-      res = [] of String
-      "bla bla ablf".scan("bl") { |s| res << s }
+      res = Array.collect "bla bla ablf".scan("bl")
```

Comment:
This particular test's purpose is to test scan with a block, so I don't think this one should change (yeah, it's using a block via the collect macro, but that obscures the test itself)